### PR TITLE
Resource Allocation Requests - ServiceBus topic notifications on request changes

### PR DESCRIPTION
**Pull request by xssic**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
Introducing fusion events, and dispatcing service bus topic notification for request changes (created and removed)

**Testing:**
- [ ] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

Tested by connecting to topic, and verifying flow to PIMS.

**Checklist:**
- [x] Considered automated tests
- [ ] Considered updating specification / documentation
- [ ] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Domain/Commands/Requests/InternalRequests/CreateInternalRequest.cs [hunk 1]
```diff
@@ -132,6 +137,9 @@ private async Task<DbResourceAllocationRequest> CreateDbRequestAsync(CreateInter
                 };
 
                 dbContext.ResourceAllocationRequests.Add(item);
+                
+                await SendNotificationsAsync(new QueryResourceAllocationRequest(item));
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Domain/Commands/Requests/InternalRequests/CreateInternalRequest.cs:141]: Should try to align this with notifications. So the domain layer do not need to worry about the event integration etc.. Just publish an event that you handle in the api project, or even the application project i guess..

File: src/backend/api/Fusion.Resources.Domain/Events/ResourceAllocationRequestEventTypes.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+
+namespace Fusion.Events
+{
+    public static class ResourceAllocationRequestEventTypes
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Domain/Events/ResourceAllocationRequestEventTypes.cs:5]: These models should not be defined in the domain model.. maybe they could be added to the Integration.Models?

**Review by HansDahle — CHANGES_REQUESTED** [review]
Good start, but i think we should try and keep the domain project as clean as we can for integration.

File: src/backend/api/Fusion.Resources.Domain/Commands/Requests/InternalRequests/CreateInternalRequest.cs (hunk 1 above)
Comment (xssic) [reply src/backend/api/Fusion.Resources.Domain/Commands/Requests/InternalRequests/CreateInternalRequest.cs]: dispatching from api project/controller

File: src/backend/api/Fusion.Resources.Domain/Events/ResourceAllocationRequestEventTypes.cs (hunk 2 above)
Comment (xssic) [reply src/backend/api/Fusion.Resources.Domain/Events/ResourceAllocationRequestEventTypes.cs]: Agree. Moved to Integration.Models

**Review by HansDahle — APPROVED** [review]

**Pull request merged**